File: 4.monitor-platform/src/main/java/com/shaolonger/monitorplatform/log/entity/JsErrorLog.java
package com.shaolonger.monitorplatform.log.entity;

import lombok.Data;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "lms_js_error_log")
@Data
public class JsErrorLog {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO, generator = "native")
    private Long id;

    private String logType;

    private Date createTime;

    private Long userId;

    private String userName;

    private String pageUrl;

    private String pageKey;

    private String deviceName;

    private String os;

    private String browserName;

    private String browserVersion;

    private String ipAddress;

    private String errorType;
    
    private String errorMessage;

    private String errorStack;
}

File: web-monitor-server/web-monitor-server-golang/model/validation/js_error_log.go
package validation

type AddJsErrorLog struct {
	LogType           string `form:"logType"`
	ProjectIdentifier string `form:"projectIdentifier"`
	Cuuid             string `form:"cUuid"`
	Buid              uint64 `form:"bUid"`
	Buname            string `form:"bUname"`
	PageUrl           string `form:"pageUrl"`
	PageKey           string `form:"pageKey"`
	DeviceName        string `form:"deviceName"`
	Os                string `form:"os"`
	OsVersion         string `form:"osVersion"`
	BrowserName       string `form:"browserName"`
	BrowserVersion    string `form:"browserVersion"`
	IpAddress         string `form:"ipAddress"`
	Address           string `form:"address"`
	NetType           string `form:"netType"`
	ErrorType         string `form:"errorType"`
	ErrorMessage      string `form:"errorMessage"`
	ErrorStack        string `form:"errorStack"`
}
